Shader program errors no longer appear for HTML5 and Android builds (and possibly other platforms)

**Describe the bug (REQUIRED)**
In Defold 1.8.1, when a fragment script was unable to compile, the console would print the error. Now, in Defold 1.9.0, the error is hidden as "Error: Unknown".

**To Reproduce (REQUIRED)**
1. Download Defold 1.9.0
2. Create a fragment program and use an unsupported function (max, pow, etc), put it in a material and have a sprite use that material or compare a float and integer in an if statement.
3. Build to HTML5 or Android

**Expected behavior (REQUIRED)**
When the fragment program is unable to be compile, it should print out the error preventing it to be compiled.

**Defold version (REQUIRED):**
  - Version 1.9.0

**Platforms (REQUIRED):**
 - Platforms: HTML5 and Android (known)
 - OS: Android, Windows 10
 - Device: Samsung Galaxy A13 5G

**Minimal repro case project (OPTIONAL):**
https://github.com/mchlkpng/shader-test

**Logs (OPTIONAL):**
How it should work:
```
INFO:ENGINE: Defold Engine 1.8.1 (fd1ad4c)
06-29 17:56:29.514 27054 27177 I defold  : INFO:DLIB: Initialized Remotery (ws://127.0.0.1:17815/rmt)
06-29 17:56:29.554 27054 27177 I defold  : INFO:ENGINE: Loading data from: archive:game.dmanifest
06-29 17:56:29.555 27054 27177 I defold  : INFO:LIVEUPDATE: Liveupdate folder located at: /data/user/0/mchlkpng.shadertest/files
06-29 17:56:29.556 27054 27177 I defold  : INFO:LIVEUPDATE: Found no legacy liveupdate zip file references
06-29 17:56:29.556 27054 27177 I defold  : INFO:LIVEUPDATE: Found no legacy liveupdate index paths
06-29 17:56:29.594 27054 27177 I defold  : INFO:ENGINE: Initialised sound device 'default'
06-29 17:56:29.675 27054 27177 W defold  : WARNING:INPUT: No gamepad map found for gamepad 0 (Virtual). The raw gamepad map will be used.
06-29 17:56:29.940 27054 27177 E defold  : ERROR:GRAPHICS: Unable to compile fragment shader.
06-29 17:56:29.940 27054 27177 E defold  : ERROR:GRAPHICS: 0:23: S0001: Type mismatch, cannot convert from 'int' to 'float'
06-29 17:56:29.940 27054 27177 W defold  : WARNING:RESOURCE: Unable to create resource: /main/materials/newsprite/newsprite.fpc: FORMAT_ERROR
06-29 17:56:29.940 27054 27177 W defold  : WARNING:RESOURCE: Unable to create resource: /main/materials/newsprite/newsprite.materialc: FORMAT_ERROR
06-29 17:56:29.940 27054 27177 W defold  : WARNING:RESOURCE: Unable to create resource: /_generated_4e94e21e40197199.spritec: FORMAT_ERROR
06-29 17:56:29.940 27054 27177 W defold  : WARNING:RESOURCE: Unable to create resource: /_generated_bc3c29d316ac2ed4.goc: FORMAT_ERROR
06-29 17:56:29.940 27054 27177 E defold  : ERROR:GAMEOBJECT: Could not instantiate game object from prototype /_generated_bc3c29d316ac2ed4.goc.
06-29 17:56:29.940 27054 27177 W defold  : WARNING:RESOURCE: Unable to create resource: /main/main.collectionc: FORMAT_ERROR
```

How it currently works:
```
06-29 18:05:24.854 29824 29888 I defold  : INFO:ENGINE: Defold Engine 1.9.0 (d6882f4)
06-29 18:05:24.950 29824 29888 I defold  : INFO:DLIB: Initialized Remotery (ws://127.0.0.1:17815/rmt)
06-29 18:05:24.965 29824 29888 I defold  : INFO:ENGINE: Loading data from: http://192.168.1.67:49765/build
06-29 18:05:24.985 29824 29888 I defold  : INFO:ENGINE: Initialised sound device 'default'
06-29 18:05:32.317 29824 29888 E defold  : ERROR:GAMESYS: Failed to create fragment program '/_generated_d996217f.fpc': Unable to compile fragment shader.
06-29 18:05:32.317 29824 29888 E defold  : Error: Unknown
06-29 18:05:32.317 29824 29888 W defold  : WARNING:RESOURCE: Unable to create resource: /_generated_d996217f.fpc: FORMAT_ERROR
06-29 18:05:32.317 29824 29888 W defold  : WARNING:RESOURCE: Unable to create resource: /main/materials/newsprite/newsprite.materialc: FORMAT_ERROR
06-29 18:05:32.317 29824 29888 W defold  : WARNING:RESOURCE: Unable to create resource: /_generated_9923214b.spritec: FORMAT_ERROR
06-29 18:05:32.318 29824 29888 W defold  : WARNING:RESOURCE: Unable to create resource: /_generated_5d98418d.goc: FORMAT_ERROR
06-29 18:05:32.318 29824 29888 E defold  : ERROR:GAMEOBJECT: Could not instantiate game object from prototype /_generated_5d98418d.goc.
06-29 18:05:32.318 29824 29888 W defold  : WARNING:RESOURCE: Unable to create resource: /main/main.collectionc: FORMAT_ERROR
06-29 18:05:32.358 29824 29888 I defold  : INFO:DLIB: SSDP: Done on address 192.168.1.140
```

**Workaround (OPTIONAL):**
Use Defold 1.8.1 if there's an issue with a shader program for different platforms

Sorry for the long wait; this is how it looks like on HTML5

```
INFO:ENGINE: Defold Engine 1.9.0 (d6882f4)
VM30:389         Made with Defold            https://www.defold.com
dmloader.js:723 INFO:ENGINE: Loading data from: archive:game.dmanifest
dmloader.js:723 INFO:LIVEUPDATE: Liveupdate folder located at: /data/.a878e2d0ce97a9d86067fbaa4676751c5a8b42f4
dmloader.js:723 INFO:LIVEUPDATE: Found no legacy liveupdate zip file references
dmloader.js:723 INFO:LIVEUPDATE: Found no legacy liveupdate index paths
VM30:4498 The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page. https://goo.gl/7K7WLu
_dmDeviceJSOpen @ VM30:4498
dmloader.js:723 INFO:ENGINE: Initialised sound device 'default'
dmloader.js:724 ERROR:GAMESYS: Failed to create fragment program '/main/materials/newsprite/newsprite.fpc': Unable to compile fragment shader.
printErr @ dmloader.js:724
dmloader.js:724 Error: Unknown
printErr @ dmloader.js:724
dmloader.js:723 WARNING:RESOURCE: Unable to create resource: /main/materials/newsprite/newsprite.fpc: FORMAT_ERROR
dmloader.js:723 WARNING:RESOURCE: Unable to create resource: /main/materials/newsprite/newsprite.materialc: FORMAT_ERROR
dmloader.js:723 WARNING:RESOURCE: Unable to create resource: /_generated_4e94e21e40197199.spritec: FORMAT_ERROR
dmloader.js:723 WARNING:RESOURCE: Unable to create resource: /_generated_bc3c29d316ac2ed4.goc: FORMAT_ERROR
dmloader.js:724 ERROR:GAMEOBJECT: Could not instantiate game object from prototype /_generated_bc3c29d316ac2ed4.goc.
printErr @ dmloader.js:724
dmloader.js:723 WARNING:RESOURCE: Unable to create resource: /main/main.collectionc: FORMAT_ERROR
dmloader.js:724 Aborted(Assertion failed: g_Window, at: ../src/platform_window_glfw.cpp,87,OnWindowFocus)
printErr @ dmloader.js:724
dmloader.js:723 INFO:CRASH: Successfully wrote Crashdump to file: /data/.Defold/_crash
dmloader.js:724 ERROR:CRASH: CALL STACK:
printErr @ dmloader.js:724
dmloader.js:724 ["RuntimeError: Aborted(Assertion failed: g_Window, at: ../src/platform_window_glfw.cpp,87,OnWindowFocus). Build with -sASSERTIONS for more info.","abort@<anonymous>:286:11","___assert_fail@<anonymous>:528:3","@wasm://wasm/00ab0ff2:wasm-function[3237]:0x16dd84","onFocusChanged@<anonymous>:8617:40","HTMLCanvasElement.onBlur@<anonymous>:8624:10"]
printErr @ dmloader.js:724
dmloader.js:724 CALL STACK END
printErr @ dmloader.js:724
dmloader.js:726 Exception thrown, see JavaScript console
VM30:287 Uncaught RuntimeError: Aborted(Assertion failed: g_Window, at: ../src/platform_window_glfw.cpp,87,OnWindowFocus). Build with -sASSERTIONS for more info.
    at abort (<anonymous>:286:11)
    at ___assert_fail (<anonymous>:528:3)
    at 00ab0ff2:0x16dd84
    at Object.onFocusChanged (<anonymous>:8617:40)
    at HTMLCanvasElement.onBlur (<anonymous>:8624:10)
```

--- a/engine/dlib/src/dlib/log.cpp
+++ b/engine/dlib/src/dlib/log.cpp
@@ -38,6 +38,9 @@
 #if defined(__APPLE__)
 #include <TargetConditionals.h>
 #endif
+#ifdef __EMSCRIPTEN__
+#include <emscripten/emscripten.h>
+#endif
 
 namespace dmLog
 {
@@ -303,11 +306,16 @@ static void DoLogPlatform(LogSeverity severity, const char* output, int output_l
 #endif
 
 #ifdef __EMSCRIPTEN__
+
         //Emscripten maps stderr to console.error and stdout to console.log.
         if (severity == LOG_SEVERITY_ERROR || severity == LOG_SEVERITY_FATAL){
-            fwrite(output, 1, output_len, stderr);
+            EM_ASM_({
+                Module.printErr(UTF8ToString($0));
+            }, output);
         } else {
-            fwrite(output, 1, output_len, stdout);
+            EM_ASM_({
+                Module.print(UTF8ToString($0));
+            }, output);
         }
 #elif !defined(ANDROID)
         fwrite(output, 1, output_len, stderr);

--- a/engine/graphics/src/opengl/graphics_opengl.cpp
+++ b/engine/graphics/src/opengl/graphics_opengl.cpp
@@ -1807,8 +1807,8 @@ static void LogFrameBufferError(GLenum status)
             glGetShaderiv(shader_id, GL_INFO_LOG_LENGTH, &logLength);
             if (logLength > 0)
             {
-                GLchar *log = (GLchar *)malloc(logLength);
-                glGetShaderInfoLog(shader_id, logLength, &logLength, log);
+                log_str = (GLchar *)malloc(logLength);
+                glGetShaderInfoLog(shader_id, logLength, &logLength, log_str);
             }
 #endif
             if (error_buffer)

